java-source/src/main/java/com/example/api/ExampleApi.java

```diff
@@ -108,14 +110,8 @@ public Response createIOU(IOU iou, @PathParam("party") String partyName) throws
 
         } catch (Throwable ex) {
             status = Response.Status.BAD_REQUEST;
-
-            if (ex.getCause() instanceof FlowSessionException) {
-                msg = "Counterparty flow terminated unexpectedly.";
-            } else if (ex.getCause() instanceof RuntimeException) {
-                msg = ex.getCause().getMessage();
-            } else {
-                msg = "Unexpected error.";
-            }
+            msg = "Transaction failed.";
+            logger.error(ex.getMessage());
```

Comment:
`logger.error("Unable to create IOU", e)` - key bit is the second param so we get the stack trace in our logs